src/control/server/config/server.go

```diff
@@ -376,41 +376,45 @@ func (c *Server) SaveToFile(filename string) error {
 }
 
 // SetPath sets the default path to the configuration file.
-func (c *Server) SetPath(inPath string) error {
-	newPath, err := common.ResolvePath(inPath, c.Path)
+func (cfg *Server) SetPath(inPath string) error {
+	newPath, err := common.ResolvePath(inPath, cfg.Path)
 	if err != nil {
 		return err
 	}
-	c.Path = newPath
+	cfg.Path = newPath
 
-	if _, err = os.Stat(c.Path); err != nil {
+	if _, err = os.Stat(cfg.Path); err != nil {
 		return err
 	}
 
 	return err
 }
 
-// SaveActiveConfig saves read-only active config, tries config dir then /tmp/
-func SaveActiveConfig(log logging.Logger, config *Server) {
-	activeConfig := filepath.Join(filepath.Dir(config.Path), configOut)
-	eMsg := "Warning: active config could not be saved (%s)"
-	err := config.SaveToFile(activeConfig)
-	if err != nil {
-		log.Debugf(eMsg, err)
+// SaveActiveConfig saves read-only active config, tries config dir then /tmp/.
+func (cfg *Server) SaveActiveConfig(log logging.Logger) {
+	activeConfig := filepath.Join(filepath.Dir(cfg.Path), configOut)
+
+	if err := cfg.SaveToFile(activeConfig); err != nil {
+		msg := "active config could not be saved"
+		log.Debug(errors.Wrap(err, msg).Error())
 
 		activeConfig = filepath.Join("/tmp", configOut)
```

Comment:
I wonder if it makes more sense to use cfg.SocketDir instead of this hard-coded /tmp value. That directory should exist and be writable by daos_server, and it would be one less place to look for stuff. Also, in theory /tmp should always be writable, but I can imagine some odd systems where it's not for whatever reason.